prov/sm2/src/sm2_msg.c

```diff
@@ -309,9 +309,9 @@ static ssize_t sm2_generic_sendmsg(struct sm2_ep *ep, const struct iovec *iov,
 	total_len = ofi_total_iov_len(iov, iov_count);
 	assert(!(op_flags & FI_INJECT) || total_len <= SM2_INJECT_SIZE);
 
-	ret = sm2_proto_ops[sm2_proto_inject](ep, peer_smr, peer_gid, op, tag,
-					      data, op_flags, FI_HMEM_SYSTEM, 0,
-					      iov, iov_count, total_len);
+	ret = sm2_proto_ops[sm2_proto_inject](
+		ep, peer_smr, peer_gid, op, tag, data, op_flags,
+		(struct ofi_mr **) desc, iov, iov_count, total_len);
```

Comment:
still confused on why clang feels the need to update this entire function call when just replacing `FI_HMEM_SYSTEM, 0` with `(struct ofi_mr **) desc` would do. looks like it does it properly on the call below

Reply:
I think it is the cast that it doesn't like

Reply:
fixed